### PR TITLE
Move fuzz from travis-ci to github actions

### PR DESCRIPTION
**Pull request by jwillemsen**

### CODE REVIEW

**Comment by jwillemsen** [comment]
See https://github.com/jwillemsen/OpenDDS/actions?query=workflow%3Afuzz

**Comment by iguessthislldo** [comment]
I am for this. The fact that linting errors can hide in timeouts on Travis has caused issues before.

**Comment by jwillemsen** [comment]
With github actions we get for free 20 concurrent jobs so this scales better (see https://help.github.com/en/actions/getting-started-with-github-actions/about-github-actions)

**Comment by jwillemsen** [comment]
For taox11 we are using already github actions for also compilation, works very good, see https://github.com/RemedyIT/taox11/actions

File: .github/workflows/fuzz.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      uses: actions/checkout@v1
+    - name: git clone dependent repositories
+      run: |
+        git clone --depth 1 git://github.com/DOCGroup/ACE_TAO.git $DOC_ROOT
+    - name: Remove not fuzzable directory
+      run: rm -rf *
+      working-directory: ./tools/modeling/plugins
+    - name: Remove not fuzzable directory
+      run: rm -rf *
+      working-directory: ./tools/rapidjson
```
Comment (mitza-oci) [inline .github/workflows/fuzz.yml]: Does this need googletest as well?  The travis script has it.

Comment (jwillemsen) [reply .github/workflows/fuzz.yml]: No, I didn't clone the submodules in this action so it is not there when running fuzz

Comment (mitza-oci) [reply .github/workflows/fuzz.yml]: In that case, can you remove rapidjson?

**Comment by mitza-oci** [comment]
What has to change in the GitHub repo configuration to get it to run the Action as part of the PR checks?

**Comment by iguessthislldo** [comment]
>What has to change in the GitHub repo configuration to get it to run the Action as part of the PR checks?

There's an "Actions" page in the repo settings with radio switch to enable them.

**Comment by mitza-oci** [comment]
> > What has to change in the GitHub repo configuration to get it to run the Action as part of the PR checks?
> 
> There's an "Actions" page in the repo settings with radio switch to enable them.

It is (and has been) set to "Enable local and third party Actions for this repository"

**Comment by jwillemsen** [comment]
I didn't had to enable anything specifically for AXCIOMA/TAOX11/ACE

**Comment by iguessthislldo** [comment]
> It is (and has been) set to "Enable local and third party Actions for this repository"

> I didn't had to enable anything specifically for AXCIOMA/TAOX11/ACE

Yeah, I think it's enabled by default, but I wanted to give the option if Adam wanted to be sure.

**Comment by jwillemsen** [comment]
Any plans to merge this?

**Comment by mitza-oci** [comment]
> Any plans to merge this?

Can we see the results of the GitHub actions here first?  Or is it just not possible?

https://github.com/objectcomputing/OpenDDS/pull/1568/checks

**Comment by jwillemsen** [comment]
No idea how, I would expect it that github would trigger a new check

**Comment by jwillemsen** [comment]
It did run om my branch, can you see it there?

**Comment by jwillemsen** [comment]
https://github.com/jwillemsen/OpenDDS/actions

**Comment by mitza-oci** [comment]
> It did run om my branch, can you see it there?

Sure, but that's not exactly what I'm asking about.  If we are using this as a Check for each PR, then this very PR that we are working on should show it passing.

Maybe we should split this up so that this PR doesn't modify travis.  Then once it's working we can come back and change travis.yml.

**Comment by mitza-oci** [comment]
Also, this may be a good opportunity to rename the script(s) since fuzz has an actual meaning in the wider software world (https://en.wikipedia.org/wiki/Fuzzing)

**Comment by jwillemsen** [comment]
See https://github.com/objectcomputing/OpenDDS/pull/1599

**Comment by iguessthislldo** [comment]
> Also, this may be a good opportunity to rename the script(s) since fuzz has an actual meaning in the wider software world (https://en.wikipedia.org/wiki/Fuzzing)

If we're not doing that here, I would like to do it in #1366.

**Pull request closed without merging**